tldw_Server_API/app/api/v1/endpoints/metrics.py

```
@@ -25,6 +27,77 @@
+_PROMETHEUS_HEADERS = {
+    "Cache-Control": "no-cache, no-store, must-revalidate",
+    "Pragma": "no-cache",
+    "Expires": "0",
+}
+_STAGE_FLAG_REFRESH_INTERVAL_SECONDS = 5.0
+_last_stage_flag_refresh = 0.0
+
+
+async def _refresh_embeddings_stage_flags() -> None:
+    """Best-effort refresh for embeddings stage flags used in text metrics export."""
+    global _last_stage_flag_refresh
```

Ruff (0.15.9)

[warning] Using the global statement to update `_last_stage_flag_refresh` is discouraged

(PLW0603)